collector/tx_processor.go

```diff
@@ -94,10 +111,40 @@ func (p *TxProcessor) Start() {
 	// start listening for transactions coming in through the channel
 	p.log.Info("Waiting for transactions...")
 	for txIn := range p.txC {
+		go p.sendTxToReceivers(txIn)
```

Comment:
this place in code is after a bunch of operations including writing to a file and request to a node

also duplicates count over all sources but we send only for allowed sources so this can lead to a big loss of txs

I think its better for receiver to do filtering then.

Reply:
You are mistaken, the transactions are written to a file later, after deduplicating, inside the processTx function. 

Your comment about loss of tx because filtering makes sense though. Would be good to add a comment about why this line of code is in this place.

Reply:
added comment https://github.com/flashbots/mempool-dumpster/pull/31/commits/62915f452c04103246c4ef4d7c8c267645bbf5e6